### ISSUE TITLE
[KV Feedback]: Deno KV release schedule?

### ISSUE THREAD

**Issue by andrefelipe** [denoland/deploy_feedback#736]
### 🔍

- [X] Did you search for existing issues?

### Type of feedback

General feedback

### Description

I'd like to rely on Deno KV for production, is everything OK with Deno KV for long term?

Do you have schedule to go out of Beta?

I already am relying on it for a client, but as the project will launch in a month I'd like to be extra sure I made the right decision. I've recently beeing burned by MongoDB Atlas Functions, where they are shutting down the entire service. I don't want to be burned twice. 

There isn't much news on Deno KV on your blog. Last post was Dec/2023

Thank you very much.

**Comment by Sleepful** [comment]
Consider this issue I raised:

https://github.com/denoland/denokv/issues/101

**Comment by andrefelipe** [comment]
Yes, I decided to not use Deno KV on production for real clients at this moment. They say it's Beta, so not their fault on this.

I confirm that I didn't see data loss in 2 months of use, with around 30k items in Deno KV. I run yesterday a code to deeply compare the data I imported (which is in a local JSON) to the live data that was in Deno KV, and I confirmed they were exact, no data loss occurred.

My show stopper was the Deno KV Backups. They look perfect on this page, https://docs.deno.com/deploy/kv/manual/backup/ but in real life to recover a backup is not friendly at all. We have to recreate a local copy of the backup and sync back to the live KV manually. I would prefer to rely on a Dashboard for that.

Anyway, was a misjudgment of mine to consider for a real client. I already switched to DynamoDB and can sleep safe now.

And on my tests DynamoDB was consistently 2x faster. So also something to test for yourself too. The database calls are more verbose, and the in-out of marshall/unmarshall is sad, but with the recent 50% price reduction is something to consider now. And we can resort to Consistent Reads when needed. https://aws.amazon.com/blogs/database/new-amazon-dynamodb-lowers-pricing-for-on-demand-throughput-and-global-tables/

Well, I will still consider DenoKV in the future, when it's out of Beta. The API is very clean, it's OK the manual way of having secondary indexes, always consistent is also nice, Date support too. Not really much to complain. I would have happily kept on Deno KV if was for the backups recoveries, no data loss confidence, and the possibility to choose the primary region.

**Comment by Sleepful** [comment]
The problem with data loss is that you may not be affected, but you might.

I agree that the main benefit to KV is their clean API and zero-config deployment.

Not sure that the distributed writes are that much of a win over a central DB with read replicas though, besides the benefit turns into a moot point if the performance is not good enough. And the zero-config becomes moot if backups require custom configuration.

EDIT: it seems foundationdb does not support distributed writes anyway, for writes it involves some level of centralization (or 'majority consensus'), which includes network latency around the cluster.

**Comment by Sleepful** [comment]
The UI to explore KV data is really poor too, AWS console is not a pleasant experience but at least it gets the job done with confidence.
